src/EventGridExtension/OutputBinding/EventGridOutputAsyncCollector.cs

```diff
@@ -21,12 +23,12 @@ public EventGridOutputAsyncCollector(EventGridAttribute attr)
             }
         }
 
-        public Task AddAsync(EventGridEvent item, CancellationToken cancellationToken = default(CancellationToken)) => _client.PublishEventsAsync(_attribute.GetTopicHostname(), new[] { item }, cancellationToken);
-
-        public Task FlushAsync(CancellationToken cancellationToken = default(CancellationToken))
+        public Task AddAsync(EventGridEvent item, CancellationToken cancellationToken = default(CancellationToken))
         {
-            // do nothing; we are sending every request as they come in to AddAsync so there's nothing to flush
+            _eventsToSend.Add(item);
             return Task.CompletedTask;
         }
+
+        public Task FlushAsync(CancellationToken cancellationToken = default(CancellationToken)) => _client.PublishEventsAsync(_attribute.GetTopicHostname(), _eventsToSend, cancellationToken);
```

Comment:
I think you need to clear the buffer after each `flush()`, usually the pattern does not limit how many times `flush()` could be called, for more info, look at [this](https://stackoverflow.com/a/40733256/1831108)

Reply:
ah, great catch. fix coming right up...